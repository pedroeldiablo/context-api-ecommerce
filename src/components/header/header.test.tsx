import React from 'react';
import ReactDOM from 'react-dom';
import { Header } from './header.component';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';


  // app.test.js
import { render, screen } from '@testing-library/react'
// import userEvent from '@testing-library/user-event'
// import { createMemoryHistory } from 'history'
import '@testing-library/jest-dom/extend-expect'


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render((
   <BrowserRouter>
    <Header />
   </BrowserRouter>), div);
 });

describe('Test suits for Header', () => {
  it('should match with snapshot', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Header/>
      </BrowserRouter>
    )
    .toJSON();
   expect(tree).toMatchSnapshot();
   });
  });

test('rendering/navigating', () => {
//   const history = createMemoryHistory()
    render(<Header/>, { wrapper: BrowserRouter })
  // verify page content for expected route
  // often you'd use a data-testid or role query, but this is also possible
  expect(screen.getByText(/CONTACT/i)).toBeInTheDocument()

//   const leftClick = { button: 0 }
//   userEvent.click(screen.getByText(/MENS/i), leftClick)

//   // check that the content changed to the new page
//   expect(screen.getByText(/you are on the about page/i)).toBeInTheDocument()
})

// test('landing on a bad page', () => {
// //   const history = createMemoryHistory()
// //   history.push('/some/bad/route')
//   render(
//     <BrowserRouter>
//      <Header/>
//    </BrowserRouter>
//   )

//   expect(screen.getByText(/no match/i)).toBeInTheDocument()
// })

// test('rendering a component that uses useLocation', () => {
//   const history = createMemoryHistory()
//   const route = '/some-route'
//   history.push(route)
//   render(
//     <Router history={history}>
//       <LocationDisplay />
//     </Router>
//   )

//   expect(screen.getByTestId('location-display')).toHaveTextContent(route)
// })
