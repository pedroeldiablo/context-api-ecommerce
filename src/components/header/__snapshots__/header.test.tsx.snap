// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for Header should match with snapshot 1`] = `
<div
  className="header"
>
  <a
    className="logo-container"
    href="/"
    onClick={[Function]}
  >
    <svg
      className="logo"
    >
      crown.svg
    </svg>
  </a>
  <div
    className="options"
  >
    <a
      className="option"
      href="/shop"
      onClick={[Function]}
    >
      SHOP
    </a>
    <a
      className="option"
      href="/contact"
      onClick={[Function]}
    >
      CONTACT
    </a>
    <a
      className="option"
      href="/signin"
      onClick={[Function]}
    >
      SIGN IN
    </a>
    <div
      className="cart-icon"
      onClick={[Function]}
    >
      <svg
        className="shopping-icon"
      >
        shopping-bag.svg
      </svg>
      <span
        className="item-count"
      >
        0
      </span>
    </div>
  </div>
</div>
`;
